Library/Homebrew/dev-cmd/extract.rb

```diff
@@ -0,0 +1,153 @@
+#:    not installed yet, attempts to install/clone the tap before continuing.
+#:    A tap must be passed through `--tap` in order for `extract` to work.
+#:
+#:    If `--force` is passed, the file at the destination will be overwritten
+#:    if it already exists. Otherwise, existing files will be preserved.
+#:
+#:    If an argument is passed through `--version`, <version> of <formula>
+#:    will be extracted and placed in the destination tap. Otherwise, the most
+#:    recent version that can be found will be used.
+
+require "utils/git"
+require "formulary"
+require "tap"
+
+class BottleSpecification
+  def method_missing(m, *_args, &_block)
+    if [:sha1, :md5].include?(m)
+      opoo "Formula has unknown or deprecated stanza: #{m}" if ARGV.debug?
+    else
+      super
+    end
+  end
+end
+
+class Module
+  def method_missing(m, *_args, &_block)
+    if [:sha1, :md5].include?(m)
+      opoo "Formula has unknown or deprecated stanza: #{m}" if ARGV.debug?
+    else
+      super
+    end
+  end
+end
+
+class DependencyCollector
+  def parse_symbol_spec(spec, tags)
+    case spec
+    when :x11        then X11Requirement.new(spec.to_s, tags)
+    when :xcode      then XcodeRequirement.new(tags)
+    when :linux      then LinuxRequirement.new(tags)
+    when :macos      then MacOSRequirement.new(tags)
+    when :arch       then ArchRequirement.new(tags)
+    when :java       then JavaRequirement.new(tags)
+    when :osxfuse    then OsxfuseRequirement.new(tags)
+    when :tuntap     then TuntapRequirement.new(tags)
+    when :ld64       then ld64_dep_if_needed(tags)
+    else
+      opoo "Unsupported special dependency #{spec.inspect}" if ARGV.debug?
+    end
+  end
+
+  module Compat
+    def parse_string_spec(spec, tags)
+      opoo "'depends_on ... => :run' is disabled. There is no replacement." if tags.include?(:run) && ARGV.debug?
+      super
+    end
+  end
+
+  prepend Compat
+end
+
+module Homebrew
+  module_function
+
+  def extract
+    Homebrew::CLI::Parser.parse do
+      flag   "--tap="
+      flag   "--version="
+      switch :debug
+      switch :force
+    end
+
+    # If no formula args are given, ask specifically for a formula to be specified
+    raise FormulaUnspecifiedError if ARGV.named.empty?
+
+    # If some other number of args are given, provide generic usage information
+    raise UsageError if ARGV.named.length != 1
+
+    odie "The tap to which the formula is extracted must be specified!" if args.tap.nil?
+
+    begin
+      formula = Formulary.factory(ARGV.named.first)
+      name = formula.name
+      repo = formula.path.parent.parent
+      file = formula.path
+    rescue FormulaUnavailableError => e
+      opoo "'#{ARGV.named.first}' does not currently exist in the core tap" if ARGV.debug?
+      core = Tap.fetch("homebrew/core")
+      name = ARGV.named.first.downcase
+      repo = core.path
+      file = core.path.join("Formula", "#{name}.rb")
```

Comment:
Prefer `/` over `.join`

Reply:
Would `core.path.join("Formula/#{name}.rb")` be all right? Or just remove it entirely, but create Pathname object around the whole thing (`Pathname.new("#{core.path}/Formula/#{name}.rb"`)? At some point, it'll need to be converted into a `Pathname` because `Git.last_revision_of_file(repo, file, before_commit: rev)` expects arguments as `Pathname` objects, not `String`.

Reply:
`core.path/"Formula/#{name}.rb"`

Reply:
(which returns a `Pathname`)